### ISSUE TITLE
Incompatibility with “Relevanssi” Plugin

### ISSUE THREAD

**Issue by VinodKKumar** [picandocodigo/List-Category-Posts#436]
[catlist search=”The Cake is a lie”]

This feature does not work with “Relevanssi” Plugin- plugin for website search. It shows empty.

After deactivating this, the { search=”The Cake is a lie” } shows results.

Relevanssi is much better than WordPress default search function in terms of results, so this cant be avoided.

Kindly make necessary changes/ suggestions/ solutions for this.

**Comment by zymeth25** [comment]
Should pin this feature request for a future update.

**Comment by VinodKKumar** [comment]
Thanks for your reply. Hope that this feature will be implemented soon.

**Comment by zymeth25** [comment]
Info on how to implement:

https://wordpress.org/support/topic/incompatibility-with-relevanssi-plugin/

**Comment by picandocodigo** [comment]
@zymeth25 just getting back into WordPress and starting to remember how this works...

If I understand correctly, we use the WP_Query object when calling `query_posts($args)` [here](https://github.com/picandocodigo/List-Category-Posts/blob/master/include/lcp-catlist.php#L91), which sets the WP_Query we want to `$wp_query`, is this right? (looking at this code after so long feels like it's the first time :laughing: ). So we would do `relevanssi_do_query( $wp_query );` inside the confitional that checks if the plugin function exists:

```php
query_posts($args);
global $wp_query;

if(function_exists('relevanssi_action_links')){
  relevanssi_do_query($wp_query);
}
```
I can write this and test it locally, but it'll be hard to add tests for integrations with other plugins.

**Comment by zymeth25** [comment]
> If I understand correctly, we use the WP_Query object when calling query_posts($args) here, which sets the WP_Query we want to $wp_query, is this right? 

Yes, 'query_posts' actually also makes it the main query. This is fine if someone wants to use LCP to generate a whole custom archive page, or a custom blog page, but should be avoided in all other cases e.g. when it's just a secondary list in a widget or part of a larger post/page. This is why the problem in #445 exists.

This also means that your solution would work but according to the docs linked in the support thread above it would run the query twice, being somewhat resource inefficient.

So perhaps the best way is to add your change and then I'll try to think of something as per #445 to improve post querying.

It's possible to write tests for this but a bit too complicated and unnecessary I think. Maybe later ;)

**Comment by zymeth25** [comment]
If it doesn't work with `query_posts` then we will first have to deal with #445 and resolve this issue later.

**Comment by noplanman** [comment]
Any progress on this? Still shows up empty when using Relevanssi
